src/pyOpenMS/tests/unittests/test_spectrum_experiment.py

```
@@ -0,0 +1,1084 @@
+    exp_repr.addSpectrum(s2)
+    exp_repr.updateRanges()
+
+    repr_str = repr(exp_repr)
+    assert "MSExperiment(" in repr_str
+    assert "num_spectra=" in repr_str
+    assert "num_chromatograms=" in repr_str
+    str_str = str(exp_repr)
+    assert str_str == repr_str
+
+
+def testSpectrumSetting(s=pyopenms.SpectrumSettings()):
```

Ruff (0.14.10)

Do not perform function call `pyopenms.SpectrumSettings` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)